server/src/main/java/io/druid/guice/StorageNodeModule.java

```diff
@@ -65,6 +65,7 @@ public DruidServerMetadata getMetadata(@Self DruidNode node, @Nullable NodeTypeC
     return new DruidServerMetadata(
         node.getHostAndPort(),
         node.getHostAndPort(),
+        node.getHostAndTlsPort(),
```

Comment:
For the first parameter here, `node.getHostAndPort()` can return null if plaintext ports are disabled but TLS ports are enabled, which can prevent nodes from announcing themselves to ZK (I saw this occur with a historical in a local test setup).